doc/design/model_zoo.md

```diff
@@ -0,0 +1,137 @@
+1. data converter, the COLUMN and LABEL clauses.
+1. model parameter file path, the path to the trained model parameters on the distributed filesystem of the cluster.
+
+It is necessary to have the model ID so users can refer to the trained model when they want to use it.  Suppose that the user typed the prediction SQL statement at the beginning of this document. SQLFlow server will convert it into a submitter program and run it with the Docker image used to train the model. Therefore, the Docker image ID is also required. The model parameter path allows the prediction submitter program to locate and load the trained models.  The data converter helps the prediction submitter to use the conversion rules consistent with the ones used when training.
+
+It is necessary to record the content or the MD5 hash of the training submitter program in the model zoo table for experiment management. Please be aware that the training submitter encodes all three categories of hyperparameters, as listed in the above sections.  Suppose that the analyst re-trains the model with different hyperparameter settings, the training submitter changes accordingly, and SQLFlow should be able to remind the analyst to either uses a new model ID or overwrites the existing row in the model zoo table.
+
+### Model Sharing
+
+After all, what is a model zoo? A model zoo refers to all the model definitions and trained models accessible by a deployment of SQLFlow.  It contains one or more model definition Docker images and the source code repositories that build the images.  It also includes the model zoo table configured to work with all submitter programs generated by the deployment.
+
+Within a deployment, it is straightforward to share a trained model.  If the analyst, `an_analyst`, in the above example wants to use her own trained model `my_first_model` for prediction, she could use the short name `my_first_model`.
+
+```sql
+SELECT ... TO PREDICT ... USING my_first_model
+```
+
+If another analyst wants to use the trained model, he would need to use the full name as `an_analyst/my_first_model`.
+
+```sql
+SELECT ... TO PREDICT ... USING an_analyst/my_first_model
+```
+
+### Model Publication
+
+Publishing a model definition, we hope all users of all SQLFlow deployments can train the model using their datasets. To achieve this goal, data scientists need to set up a continuous deployment system to automatically build their model definition repositories into Docker images and push to a public Docker registry like DockerHub.com.  In the above example, the Docker image `a_data_scientist/regressors` is one hosted on DockerHub.com.
+
+It requires more steps to publish a trained model. We need a public registry like DockerHub.com, but hosts rows from the model zoo table and corresponding model parameters.  We propose the following extended SQL syntax for analysts `an_analyst` to publish her trained model.
+
+```sql
+SQLFLOW PUBLISH my_first_model
```

Comment:
Can we also use the syntax like `SQLFLOW LIST` to list all models available and `SQLFLOW DESCRIBE my_first_model` to show the model's document?

Reply:
Good idea! I might not have sufficient time today to follow your suggestion in this PR. Please feel free to merge this PR and submit successive PRs to complete these ideas.